srtcore/api.cpp

```diff
@@ -1679,13 +1679,8 @@ int CUDTUnited::close(CUDTSocket* s)
            }
 
            HLOGC(mglog.Debug, log << "@" << u << " GLOBAL CLOSING: ... still waiting for any update.");
-           CTimer::EWait wt = CTimer::waitForEvent();
-
-           if ( wt == CTimer::WT_ERROR )
-           {
-               HLOGC(mglog.Debug, log << "GLOBAL CLOSING: ... ERROR WHEN WAITING FOR EVENT. Exiting close() to prevent hangup.");
-               break;
-           }
```

Comment:
Maybe here the pthread version should throw an exception in case of an error

Reply:
If it's in the same situation as in case of C++11, then sure. There would be some common wrapper needed for the exception type name.

Reply:
Added in TODO of #1103.